app/models/vm_migrate_task.rb

```diff
@@ -66,18 +71,20 @@ def do_request
     begin
       if vc_method == :migrate
         vm.migrate(host, respool)
-      else
+      elsif vc_method == :relocate
         vm.relocate(host, respool, datastore, nil, disk_transform)
       end
+
+      folder.move_into_folder(vm) if folder.present?
```

Comment:
If you move this into the provider and *don't* have it at the base, this can be `folder.try(:move_into_folder, vm)`